DPPP/test/unit/tPSet.cc

```diff
@@ -0,0 +1,447 @@
+// tPSet.cc: Test program for class PreFlagger::PSet
+// Copyright (C) 2010
+// ASTRON (Netherlands Institute for Radio Astronomy)
+// P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
+//
+// This file is part of the LOFAR software suite.
+// The LOFAR software suite is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The LOFAR software suite is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
+//
+// $Id$
+//
+// @author Ger van Diepen
+
+#include "../../PreFlagger.h"
+#include "../../DPInfo.h"
+#include "../../../Common/ParameterSet.h"
+
+#include <casacore/casa/Arrays/ArrayIO.h>
+#include <casacore/casa/Quanta/MVTime.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+
+namespace {
+
+// Simple class to generate input arrays.
+// It can only set all flags to true or all to false.
+// Weights are always 1.
+// It can be used with different nr of times, channels, etc.
+class TestInput: public DP3::DPPP::DPInput
+{
+public:
+  TestInput(int nbl, int nchan, int ncorr)
+    : itsNChan(nchan), itsNCorr(ncorr)
+  {
+    info().init (itsNCorr, 0, itsNChan, 0, 0, 50, string(), string());
+    // Fill the baseline stations; use 4 stations.
+    // So they are called 00 01 02 03 10 11 12 13 20, etc.
+    casacore::Vector<casacore::Int> ant1(nbl);
```

Comment:
We don't have to use casacore::Int-like types (also casacore::String,  uInt, etc.) -- they are an artefact (and, except for String, I expect are to be deprecated in casacore at some point). Where possible we also shouldn't use a casacore::Vector, because they are a) much slower than a std::vector; and b) have really confusing copy/reference semantics, but  we can save cleaning that up the latter for another time ;).